static/js/src/renewal-modal.js

```diff
@@ -1,469 +1,708 @@
+    return true;
+  }
 
-    cardImgEl.innerHTML = cardInfo.brand;
-    cardTextEl.innerHTML = cardText;
-    cardExpiryEl.innerHTML = cardExpiry;
-    customerNameEl.innerHTML = billingInfo.name;
-    customerEmailEl.innerHTML = billingInfo.email;
+  return false;
+}
+
+function resetModal() {
+  form.reset();
+  card.clear();
+  resetProgressIndicator();
+  modal.classList.remove("is-dialog-mode", "is-pay-mode");
+  modal.classList.add("is-details-mode");
+  addPaymentMethodButton.disabled = true;
+  processPaymentButton.disabled = true;
+
+  customerInfo = {
+    name: null,
+    email: null,
+    country: null,
+    address: null,
+    tax: null,
+  };
+}
+
+function resetProgressIndicator() {
+  progressIndicator
+    .querySelector(".p-icon--spinner")
+    .classList.remove("u-hide");
+  progressIndicator.querySelector(".p-icon--success").classList.add("u-hide");
+  progressIndicator.querySelector("span").innerHTML = "";
+  progressIndicator.classList.add("u-hide");
+}
+
+function sendGAEvent(label) {
+  dataLayer.push({
+    event: "GAEvent",
+    eventCategory: "advantage",
+    eventAction: "renewal",
+    eventLabel: label,
+    eventValue: undefined,
+  });
+}
+
+function setupCardElements() {
+  card.mount("#card-element");
+
+  card.on("change", (event) => {
+    if (event.error) {
+      const errorObject = parseForErrorObject(event.error);
+      cardValid = false;
+      addPaymentMethodButton.disabled = true;
+      presentError(errorObject);
+    } else if (event.complete) {
+      cardValid = true;
+      hideErrors();
+      validateForm();
+    }
+  });
+}
+
+function showDetailsMode() {
+  disableProcessingState();
+  modal.classList.remove("is-pay-mode", "is-dialog-mode");
+  modal.classList.add("is-details-mode");
+  processPaymentButton.disabled = true;
+  card.focus();
+  validateForm();
+}
+
+function showDialogMode() {
+  disableProcessingState();
+  modal.classList.remove("is-pay-mode", "is-details-mode");
+  modal.classList.add("is-dialog-mode");
+  processPaymentButton.disabled = true;
+  processPaymentButton.disabled = true;
+}
+
+function showPayMode() {
+  hideErrors();
+  disableProcessingState();
+  modal.classList.remove("is-details-mode", "is-dialog-mode");
+  modal.classList.add("is-pay-mode");
+  addPaymentMethodButton.disabled = true;
+  processPaymentButton.disabled = false;
+}
+
+function toggleModal() {
+  if (modal && modal.classList.contains("p-modal")) {
+    modal.classList.toggle("u-hide");
+    document.documentElement.classList.toggle("p-modal--active");
   }
+}
 
-  function setupCardElements() {
-    card.mount("#card-element");
+function validateForm() {
+  const inputs = form.elements;
+  let inputsValidity = [cardValid];
 
-    card.on("change", ({ error }) => {
-      if (error) {
-        presentError(error.message, "card");
-      } else {
-        hideErrors();
-      }
-      addPaymentMethodButton.disabled = false;
-    });
+  for (let i = 0; i < inputs.length; i++) {
```

Comment:
could you use `map` here instead?

Reply:
would have been nicer, but map isn't available on a HTMLFormControlsCollection